etna/records/models.py

```diff
@@ -1,78 +1,319 @@
+        """
+        Returns `True` if a 'reference_number' value can be extracted from the
+        raw data for this record. Otherwise `False`.
+        """
+        try:
+            self.reference_number
+        except ValueExtractionError:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def url(self):
+        """
+        Return the "url" value for this record. This value is typically
+        only present for 'interpretive' results from other websites.
+
+        Raises `ValueExtractionError` when the raw data does not include
+        values in any of the expected positions.
+        """
+        try:
+            return self.template["sourceUrl"]
+        except KeyError:
+            raise ValueExtractionError(
+                f"'url' could not be extracted from source data: {self._raw}"
+            )
+
+    def has_url(self) -> bool:
+        """
+        Returns `True` if a 'url' value can be extracted from the raw data
+        for this record. Otherwise `False`.
+        """
+        try:
+            self.url
+        except ValueExtractionError:
+            return False
+        else:
+            return True
+
+    @cached_property
+    def title(self) -> str:
+        try:
+            return self.template["summaryTitle"]
+        except KeyError:
+            return self.get("summary.title", default="")
+
+    @cached_property
+    def closure_status(self) -> str:
+        try:
+            return self.template["accessCondition"]
+        except KeyError:
+            self.get("availability.access.condition.value", default="")
+        return
+
+    @cached_property
+    def arrangement(self) -> str:
+        try:
+            raw = self.template["arrangement"]
+        except KeyError:
+            raw = self.get("arrangement.value", default="")
+        return format_description_markup(raw)
+
+    @cached_property
+    def legal_status(self) -> str:
+        try:
+            return self.template["legalStatus"]
+        except KeyError:
+            return self.get("legal.status", default="")
+
+    @cached_property
+    def is_digitised(self) -> bool:
+        return self.get("digitised", default=False)
+
+    @cached_property
+    def availability_delivery_surrogates(self) -> str:
+        return self.get("availability.delivery.surrogate", default="")
+
+    @cached_property
+    def media_reference_id(self) -> str:
+        return self.get("multimedia.@admin.id", default="")
+
+    @cached_property
+    def catalogue_source(self) -> str:
+        return self.get("source.value", default="")
+
+    @cached_property
+    def description(self) -> str:
+        descriptions = self.get("description", default=())
+        if len(descriptions) == 1:
+            return format_description_markup(descriptions[0]["value"])
+        for item in descriptions:
+            if item.get("type", "") == "description":
+                return format_description_markup(item["value"])
```

Comment:
We have `@template.details.description` available to use, perhaps as default, but I'm not sure if it picks up the correct description when there is more than one type="description". With len=1, the type = "scope and content" so far as I seen.

Reply:
Have decided to ditch that code and just use `highlight.@template.details.description` or `@template.details.description`